testsuite/objects/__init__.py

```diff
@@ -99,6 +99,33 @@ class Rule:
     value: str
 
 
+@dataclass
+class AnyPatterns:
+    """Dataclass specifying *OR* operation on patterns. Any one needs to pass for this block to pass."""
+
+    any: list["Patterns"]
+
+
+@dataclass
+class AllPatterns:
+    """Dataclass specifying *AND* operation on patterns. All need to pass for this block to pass."""
+
+    all: list["Patterns"]
```

Comment:
can you have `all` and `any` defined at the same time?

Reply:
Yes, you can define them alongside each other or inside each other. Look at the example I created for the `test_patternref_expressions.py` test. There exists default "All" at the top of rules list. Look at ii) in https://github.com/Kuadrant/authorino/blob/main/docs/features.md#common-feature-conditions-when